cabal-install/src/Distribution/Client/Setup.hs

```diff
@@ -1541,6 +1543,55 @@ genBoundsCommand =
         ]
     }
 
+-- ------------------------------------------------------------
+-- Check command
+-- ------------------------------------------------------------
+
+data CheckFlags = CheckFlags
+  { checkVerbosity :: Flag Verbosity
+  , checkIgnore :: [CheckExplanationIDString]
+  }
+  deriving (Show, Generic, Typeable)
+
+defaultCheckFlags :: CheckFlags
+defaultCheckFlags =
+  CheckFlags
+    { checkVerbosity = Flag normal
+    , checkIgnore = []
+    }
+
+checkCommand :: CommandUI CheckFlags
+checkCommand =
+  CommandUI
+    { commandName = "check"
+    , commandSynopsis = "Check the package for common mistakes."
+    , commandDescription = Just $ \_ ->
+        wrapText $
+          "Expects a .cabal package file in the current directory.\n"
+            ++ "\n"
+            ++ "The checks correspond to the requirements to packages on Hackage. "
+            ++ "If no errors and warnings are reported, Hackage will accept this "
```

Comment:
A little surprising that this patch changes the error message (to not mention the exit code anymore).

Reply:
Sorry, not error message, but help message.

Reply:
Well spotted, careless rebase.